pkg/devfile/parse.go

```diff
@@ -0,0 +1,62 @@
+package devfile
+
+import (
+	"github.com/devfile/parser/pkg/devfile/parser"
+	// "github.com/devfile/parser/pkg/devfile/validate"
+)
+
+// This is the top level parse code which has validation code specific to odo hence it cannot be kept inside the
+// devfile/parser package. That package is supposed to be independent of odo.
+
+// ParseFromURLAndValidate func parses the devfile data from the url
+// and validates the devfile integrity with the schema
+// and validates the devfile data.
+// Creates devfile context and runtime objects.
+func ParseFromURLAndValidate(url string) (d parser.DevfileObj, err error) {
```

Comment:
The functions in this file are just wrappers on `parser.Parse[...]()` *without* validation. I'm not clear why they should exist (and why consumers wouldn't just do parsing and validation in two steps.

Reply:
the validation will be uncommented once we have it ready in devfile/api

Reply:
Makes sense, thanks.